### PR TITLE
allow the user to provide their own zstd dependency

### PR DESCRIPTION
**Pull request by robertu94**
This is needed for spack support.  Apparently we didn't get all of my fixes in when we switched from my repo to yours. I'll add explanatory comments below.  I generally tried to default to what the behavior was before I touched anything.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,6 +35,9 @@ option( BUILD_UNIT_TESTS "Build unit tests using GoogleTest" ON )
 option( BUILD_CLI_UTILITIES "Build a set of command line utilities" ON )
 option( USE_OMP "Use OpenMP parallelization on 3D volumes" ON )
 option( USE_ZSTD "Incorporate ZSTD to achieve further reduction (~5%)" OFF )
+cmake_dependent_option(USE_BUNDLED_ZSTD "prefer an the bundled ZStd to an external one" ON "USE_ZSTD" OFF)
```
Comment (robertu94) [inline CMakeLists.txt:38]: Spack requires using a dependencies from spack when they are installed.  This is so the user can customize the build of them

File: CMakeLists.txt [hunk 2]
```diff
@@ -74,10 +78,11 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR})
 if( USE_ZSTD )
     message(STATUS "ZSTD add-on enabled!")
 
+    if(USE_BUNDLED_ZSTD)
     project(zstd-download NONE)
     include(ExternalProject)
     ExternalProject_Add( zstd-download
-      GIT_REPOSITORY    git@github.com:facebook/zstd.git
+      GIT_REPOSITORY    https://github.com/facebook/zstd.git
```
Comment (robertu94) [inline CMakeLists.txt:85]: using a `git@` url causes build scripts to hang if the user hasn't configured password-less SSH keys for github. 😞 I wish we could just use git here.

File: CMakeLists.txt [hunk 3]
```diff
@@ -105,11 +110,18 @@ if( USE_ZSTD )
     ExternalProject_Get_Property( zstd-download INSTALL_DIR )
     include_directories("${INSTALL_DIR}/include")
     link_directories( "${INSTALL_DIR}/lib" )
+    set(SPERR_ZSTD_DEPENDENCY "")
+    else()
+      find_package(PkgConfig REQUIRED)
+      pkg_search_module(ZSTD IMPORTED_TARGET GLOBAL libzstd REQUIRED)
+      set(SPERR_ZSTD_DEPENDENCY "libzstd")
```
Comment (robertu94) [inline CMakeLists.txt:117]: SPERR_ZSTD_DEPENDENCY is used in `SPERR.pc` to link the external libzstd when provided by the user.

File: CMakeLists.txt [hunk 4]
```diff
@@ -105,11 +110,18 @@ if( USE_ZSTD )
...
+    endif()
 else()
     message(STATUS "ZSTD add-on disabled!")
 endif()
 
 
+if(USE_SPERR_PREFERED_RPATH)
```
Comment (robertu94) [inline CMakeLists.txt:124]: SPACK set's it own RPATH settings which are getting overwritten here.  Provide a way to turn these off.

**Review by shaomeng — APPROVED** [review]
This change looks really cool, thanks Robert!

**Pull request merged**